src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -447,11 +446,11 @@ private void NotifyRenameEventArgs(WatcherChangeTypes action, ReadOnlySpan<char>
         /// </summary>
         private void NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySpan<char> name)
         {
-            FileSystemEventHandler? handler = GetHandler(changeType);
+            Action<FileSystemEventArgs>? raiser = GetRaiser(changeType);
 
-            if (handler != null && MatchPattern(name.IsEmpty ? _directory : name))
+            if (raiser != null && MatchPattern(name.IsEmpty ? _directory : name))
             {
-                handler(this, new FileSystemEventArgs(changeType, _directory, name.IsEmpty ? null : name.ToString()));
+                raiser(new FileSystemEventArgs(changeType, _directory, name.IsEmpty ? null : name.ToString()));
```

Comment:
These changes will result in delegates being allocated for every event raised, which is not something we want to do.  It's also unnecessary.

This method can be reverted (along with the changes to GetHandler), and then this one line changed to just be:
```C#
InvokeOn(new FileSystemEventArgs(changeType, _directory, name.IsEmpty ? null : name.ToString()), handler);
```

Reply:
I have made the change you suggested and reverted, except I have removed the `handler != null` check, since the check is also in `InvokeOn()`.

Reply:
Reverting means that `OnCreated()`, `OnChanged()` and `OnDeleted()` are not used by `FileSystemWatcher`. Would you prefer it if I called `CallNotifyFileSystemEventArgs()` instead of these methods, in the unit tests such as this:

https://github.com/dotnet/runtime/blob/fda04b9d36c273e3fcef1fe911d8502de944125a/src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs#L324-L345

Reply:
> I have made the change you suggested and reverted, except I have removed the handler != null check, since the check is also in InvokeOn().

Please put back the null checks.  We don't want to do work like allocating the event args or performing a name match if there's no handler registered.  The extra null check won't matter.

Reply:
> Would you prefer it if I called CallNotifyFileSystemEventArgs() instead of these methods, in the unit tests such as this:

No, these are tests of the protected methods; they're part of the exposed surface area and we need to continue to validate them even if we don't use them ourselves.